store/v2/pruning/manager.go

```diff
@@ -67,3 +72,47 @@ func (m *Manager) SignalCommit(start bool, version uint64) error {
 
 	return nil
 }
+
+// PruneKVStores prunes KVStores which are needed to be pruned by upgrading the
+// store key.
+func (m *Manager) PruneKVStores(kvStores []corestore.KVStoreWithBatch) error {
+	if len(kvStores) == 0 {
+		return nil
+	}
+
+	for _, kvStore := range kvStores {
+		iter, err := kvStore.Iterator(nil, nil)
+		if err != nil {
+			return err
+		}
+		defer iter.Close()
```

Comment:
info: if you expect a larger number of iterators being used, it can make sense to wrap the usage into a closure so that they don't pile up